main.go

```diff
@@ -22,47 +24,57 @@ import (
 	"github.com/goadesign/goa"
 	"github.com/goadesign/goa/middleware"
 	"github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/spf13/viper"
 )
 
 var (
 	// Commit current build commit set by build script
 	Commit = "0"
 	// BuildTime set by build script
 	BuildTime = "0"
-	// Development enables certain dev only features, like auto token generation
-	Development = false
-)
-
-const (
-	// DBMaxRetryAttempts is the number of times alm server will attempt to open a connection to the database before it gives up
-	DBMaxRetryAttempts int = 50
 )
 
 func main() {
-	printUserInfo()
-
-	var dbHost string
+	// --------------------------------------------------------------------
+	// Parse flags
+	// --------------------------------------------------------------------
+	var configFilePath string
+	var printConfig bool
 	var scheduler *remoteworkitem.Scheduler
-
-	flag.BoolVar(&Development, "dev", false, "Enable development related features, e.g. token generation endpoint")
-	flag.StringVar(&dbHost, "dbhost", "", "The hostname of the db server")
+	flag.StringVar(&configFilePath, "config", configuration.DefaultConfigFilePath, "Path to the config file to read")
+	flag.BoolVar(&printConfig, "printConfig", false, "Prints the config (including merged environment variables) and exits")
 	flag.Parse()
 
-	if len(dbHost) == 0 {
-		dbHost = os.Getenv("DBHOST")
+	// Override -config switch with environment variable if provided.
+	if envConfigPath, ok := os.LookupEnv("ALMIGHTY_CONFIG_FILE_PATH"); ok {
```

Comment:
Command line parameters should override env vars, not the other way around. Otherwise you'd have to unset the env var each time you want to override a setting.